<h1 align="center">Online Retail Data</h1>

In [29]:
import kagglehub
import pandas as pd
import numpy as np
import os

#Load the dataset
retail_sales_df = pd.read_csv("../Dataset/online_retail_sales_dataset.csv")
retail_sales_df.head()

transaction_id            timestamp  customer_id  product_id  \
0               1  2023-01-01 00:00:00         1993         915   
1               2  2023-01-01 00:01:00         3474         553   
2               3  2023-01-01 00:02:00         4564         248   
3               4  2023-01-01 00:03:00         1133         948   
4               5  2023-01-01 00:04:00         3626         284   

         product_category  quantity   price  discount payment_method  \
0          Home & Kitchen         8  103.30      0.23      Gift Card   
1                Clothing         9  180.28      0.31      Gift Card   
2  Beauty & Personal Care         7   81.58      0.27     Debit Card   
3                Clothing         3  235.20      0.00     Debit Card   
4                   Books         9  453.00      0.34    Credit Card   

   customer_age customer_gender customer_location  total_amount  
0            27          Female     North America        636.33  
1            53           Other     South America       1119.54  
2            34           Other     North America        416.87  
3            50           Other         Australia        705.60  
4            23          Female         Australia       2690.82

In [30]:
retail_sales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 13 columns):
 #   Column             Non-Null Count    Dtype  
---  ------             --------------    -----  
 0   transaction_id     1000000 non-null  int64  
 1   timestamp          1000000 non-null  object 
 2   customer_id        1000000 non-null  int64  
 3   product_id         1000000 non-null  int64  
 4   product_category   1000000 non-null  object 
 5   quantity           1000000 non-null  int64  
 6   price              1000000 non-null  float64
 7   discount           1000000 non-null  float64
 8   payment_method     1000000 non-null  object 
 9   customer_age       1000000 non-null  int64  
 10  customer_gender    1000000 non-null  object 
 11  customer_location  1000000 non-null  object 
 12  total_amount       1000000 non-null  float64
dtypes: float64(3), int64(5), object(5)
memory usage: 99.2+ MB


>  - 100,000 Transactions      
>  - 13 features       
>  - Seemingly no null values      
>  - timestamp Datatype needs to be changed


In [31]:
retail_sales_df.describe()

transaction_id     customer_id      product_id        quantity  \
count  1000000.000000  1000000.000000  1000000.000000  1000000.000000   
mean    500000.500000     2998.854163      549.804766        5.003086   
std     288675.278932     1154.764311      259.848068        2.581004   
min          1.000000     1000.000000      100.000000        1.000000   
25%     250000.750000     1999.000000      325.000000        3.000000   
50%     500000.500000     2999.000000      550.000000        5.000000   
75%     750000.250000     3998.000000      775.000000        7.000000   
max    1000000.000000     4999.000000      999.000000        9.000000   

                price        discount    customer_age    total_amount  
count  1000000.000000  1000000.000000  1000000.000000  1000000.000000  
mean       252.445705        0.250204       43.508625      946.839858  
std        142.868633        0.144384       15.007346      810.902205  
min          5.000000        0.000000       18.000000        2.560000  
25%        128.690000        0.130000       31.000000      297.180000  
50%        252.700000        0.250000       44.000000      715.710000  
75%        375.990000        0.380000       57.000000     1397.750000  
max        500.000000        0.500000       69.000000     4496.130000

> Mean Quatities: 5 (On average 5 items per transactions)       
> Mean Price : $ 252 (On Average $250 per transactions)

In [32]:
retail_sales_df.isnull().sum()

transaction_id       0
timestamp            0
customer_id          0
product_id           0
product_category     0
quantity             0
price                0
discount             0
payment_method       0
customer_age         0
customer_gender      0
customer_location    0
total_amount         0
dtype: int64

> No null values

In [33]:
#Check Duplicates
retail_sales_df.duplicated().sum()

0

> No duplicates. 